Show recently completed pull requests matching the settings

jornletnes opened the pull request:
Show recently completed pull requests that match the monitor settings in an expander underneath the approved pull requests. Unlike in the "Approval needed" and "Approved" lists, the PRs in the completed list view are shown youngest to oldest (where age is judged by id).

Also in this pull request are changes to make it easier to disable the strong-name signing and instructions on how to do so as a workaround to keep Visual Studio responsive.

Commit:
updated third party components

Commit:
added capability to show completed pull requests

Commit:
invert sort order, to make the newest pull request show up at the top of the list

nickform commented on PullRequestMonitor.UnitTest/Model/TfProjectTest.cs:
```diff
@@ -376,6 +376,19 @@ public void TestApprovedPullRequestCount_ReturnsApprovedCount(int numApprovedPul
         }
 
+        [Test, TestCaseSource(nameof(PullRequestCountCases))]
+        public void TestCompletedPullRequestCount_ReturnCompletedCount(int numCompletedPullRequests)
```
Minor: ```_ReturnsCompletedCount```.

nickform commented on PullRequestMonitor/Model/ITfProject.cs:
```diff
@@ -0,0 +1,29 @@
+﻿using System;
```
Thanks for pulling this out [hangs head].

nickform commented on PullRequestMonitor/Model/TfProject.cs:
```diff
@@ -115,11 +98,14 @@ public void DoPullRequestRetrieval()
 
                 var activePullRequestIds = EnsureActivePrsArePresent(getPRsTask.Result);
                 RemoveStalePullRequests(activePullRequestIds);
+                EnsureCompletedPrsArePresent(getCompletedPullRequestsTask.Result);
```
Would be great to do this only when needed if it doesn't complicate things too badly. I think the MVVM is okay in this area so hopefully not.

nickform commented on PullRequestMonitor/TfsConnection.cs:
```diff
@@ -65,11 +71,21 @@ public Task<IEnumerable<ITfGitRepository>> GetRepositoriesInProject(ITfProject p
+            return PullRequestsInProject(project, CompletedPullRequestSearchCriteria);
+        }
+
+        private Task<IEnumerable<IPullRequest>> PullRequestsInProject(ITfProject project, GitPullRequestSearchCriteria pullRequestSearchCriteria)
```
Nice. Thanks.

Commit:
un-upgrade third party components

nickform commented:
Looking good. Definitely think we should back out the dependency upgrades and reverse the sort order for the completed pull requests. Also, the scroll bar is gone for me when I build this. Is it there for you?

nickform commented:
For some reason the pull request validation build is not triggering for forks even though there is a settings which says "Build pull requests from forks of this repository" which I have enabled. I changed one other setting. Let's see if it fires the next time you update the branch?

Commit:
make the Completed changelists show up in opposite order (newest on top)

nickform commented:
Damnit - still no build!

nickform commented:
Well I got the build to run but only by creating an identical branch with its own pull request in my repo.

nickform commented on PullRequestMonitor/View/SingleProjectView.xaml:
```diff
@@ -21,5 +22,13 @@
             </StackPanel>
         </ScrollViewer>
+        <ScrollViewer Grid.Row="2" VerticalScrollBarVisibility="Auto">
+            <Expander IsExpanded="True">
```
This needs to move into the other ScrollViewer otherwise the scroll bar doesn't appear when the active pull requests overflow the window space (and then you can't see the completed pull requests at all).

nickform committed:
Fix missing scrollbar and window test

Commit:
test for pull request descending list vievmodel class

Commit:
Document a workaround for Visual Studio slowness

- Document the workaround in CONTRIBUTING.md
- Make everything in the production assembly public so that InternalsVisibleTo isn't needed.

Commit:
Pass scroll wheel through the list views

nickform reviewed (approved):
I fixed the scroll bar issue and Jørn fixed the sort order so think this is good to merge.
